<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>645</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <property name="spacing">
     <number>1</number>
    </property>
    <property name="leftMargin">
     <number>1</number>
    </property>
    <property name="topMargin">
     <number>1</number>
    </property>
    <property name="rightMargin">
     <number>1</number>
    </property>
    <property name="bottomMargin">
     <number>1</number>
    </property>
    <item>
     <widget class="CodeEditor" name="textEdit"/>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>文件(&amp;F)</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionNew"/>
    <addaction name="actionSave"/>
    <addaction name="actionSaveAs"/>
   </widget>
   <widget class="QMenu" name="menu_E">
    <property name="title">
     <string>编辑(&amp;E)</string>
    </property>
    <addaction name="actionCut"/>
    <addaction name="actionUndo"/>
    <addaction name="actionCopy"/>
    <addaction name="actionPaste"/>
   </widget>
   <widget class="QMenu" name="menu_O">
    <property name="title">
     <string>格式(&amp;O)</string>
    </property>
    <addaction name="actionFont"/>
    <addaction name="actionWrap"/>
    <addaction name="actionListNumber"/>
    <addaction name="actionFontcolor"/>
    <addaction name="actionbackcolor"/>
    <addaction name="actionfgbclolor"/>
   </widget>
   <widget class="QMenu" name="menu_V">
    <property name="title">
     <string>查看(&amp;V)</string>
    </property>
    <addaction name="actionStatusBar"/>
    <addaction name="actionToolBar"/>
   </widget>
   <widget class="QMenu" name="menu_H">
    <property name="title">
     <string>帮助(&amp;H)</string>
    </property>
    <addaction name="actionFind"/>
    <addaction name="actionReplace"/>
    <addaction name="actionAbout"/>
    <addaction name="actionExit"/>
   </widget>
   <addaction name="menu"/>
   <addaction name="menu_E"/>
   <addaction name="menu_O"/>
   <addaction name="menu_V"/>
   <addaction name="menu_H"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionNew"/>
   <addaction name="actionOpen"/>
   <addaction name="actionSave"/>
   <addaction name="actionSaveAs"/>
   <addaction name="separator"/>
   <addaction name="actionUndo"/>
   <addaction name="actionCut"/>
   <addaction name="actionCopy"/>
   <addaction name="actionPaste"/>
   <addaction name="separator"/>
   <addaction name="actionFind"/>
   <addaction name="actionReplace"/>
   <addaction name="separator"/>
   <addaction name="actionFont"/>
   <addaction name="actionWrap"/>
   <addaction name="separator"/>
   <addaction name="actionToolBar"/>
   <addaction name="action_redo"/>
   <addaction name="actionStatusBar"/>
   <addaction name="actionSelectAll"/>
   <addaction name="actionListNumber"/>
   <addaction name="actionAddBookmark"/>
   <addaction name="actionShowBookmarks"/>
  </widget>
  <action name="actionNew">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/img/newfile_FILL0_wght400_GRAD0_opsz24.png</normaloff>:/img/newfile_FILL0_wght400_GRAD0_opsz24.png</iconset>
   </property>
   <property name="text">
    <string>新建(&amp;N)</string>
   </property>
   <property name="toolTip">
    <string>新建</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/img/folder_open_FILL0_wght400_GRAD0_opsz24.png</normaloff>:/img/folder_open_FILL0_wght400_GRAD0_opsz24.png</iconset>
   </property>
   <property name="text">
    <string>打开(&amp;O)</string>
   </property>
   <property name="toolTip">
    <string>打开</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/img/save_FILL0_wght400_GRAD0_opsz24.png</normaloff>:/img/save_FILL0_wght400_GRAD0_opsz24.png</iconset>
   </property>
   <property name="text">
    <string>保存(&amp;S)</string>
   </property>
   <property name="toolTip">
    <string>保存</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSaveAs">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/img/save_as_FILL0_wght400_GRAD0_opsz24.png</normaloff>:/img/save_as_FILL0_wght400_GRAD0_opsz24.png</iconset>
   </property>
   <property name="text">
    <string>另存为</string>
   </property>
   <property name="toolTip">
    <string>另存为</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>退出(&amp;X)</string>
   </property>
   <property name="toolTip">
    <string>退出</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+X</string>
   </property>
  </action>
  <action name="actionUndo">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/img/undo_FILL0_wght400_GRAD0_opsz24.png</normaloff>:/img/undo_FILL0_wght400_GRAD0_opsz24.png</iconset>
   </property>
   <property name="text">
    <string>撤销(&amp;Z)</string>
   </property>
   <property name="toolTip">
    <string>撤销</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
  </action>
  <action name="actionCut">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/img/cut_FILL0_wght400_GRAD0_opsz24.png</normaloff>:/img/cut_FILL0_wght400_GRAD0_opsz24.png</iconset>
   </property>
   <property name="text">
    <string>剪切(&amp;X)</string>
   </property>
   <property name="toolTip">
    <string>剪切</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+X</string>
   </property>
  </action>
  <action name="actionCopy">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/img/folder_copy_FILL0_wght400_GRAD0_opsz24.png</normaloff>:/img/folder_copy_FILL0_wght400_GRAD0_opsz24.png</iconset>
   </property>
   <property name="text">
    <string>复制(&amp;C)</string>
   </property>
   <property name="toolTip">
    <string>复制</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="actionPaste">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/img/content_paste_FILL0_wght400_GRAD0_opsz24.png</normaloff>:/img/content_paste_FILL0_wght400_GRAD0_opsz24.png</iconset>
   </property>
   <property name="text">
    <string>粘贴(&amp;V)</string>
   </property>
   <property name="toolTip">
    <string>粘贴</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+V</string>
   </property>
  </action>
  <action name="actionFind">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/img/search_FILL0_wght400_GRAD0_opsz24.png</normaloff>:/img/search_FILL0_wght400_GRAD0_opsz24.png</iconset>
   </property>
   <property name="text">
    <string>查找(&amp;F)</string>
   </property>
   <property name="toolTip">
    <string>查找</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+F</string>
   </property>
  </action>
  <action name="actionReplace">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/img/find_replace_FILL0_wght400_GRAD0_opsz24.png</normaloff>:/img/find_replace_FILL0_wght400_GRAD0_opsz24.png</iconset>
   </property>
   <property name="text">
    <string>替换(&amp;H)</string>
   </property>
   <property name="toolTip">
    <string>替换</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+H</string>
   </property>
  </action>
  <action name="actionFont">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/img/text_fields_FILL0_wght400_GRAD0_opsz24.png</normaloff>:/img/text_fields_FILL0_wght400_GRAD0_opsz24.png</iconset>
   </property>
   <property name="text">
    <string>字体(&amp;D)</string>
   </property>
   <property name="toolTip">
    <string>字体</string>
   </property>
   <property name="shortcut">
    <string>Alt+D</string>
   </property>
  </action>
  <action name="actionWrap">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/img/wrap_text_FILL0_wght400_GRAD0_opsz24.png</normaloff>:/img/wrap_text_FILL0_wght400_GRAD0_opsz24.png</iconset>
   </property>
   <property name="text">
    <string>自动换行</string>
   </property>
   <property name="toolTip">
    <string>自动换行</string>
   </property>
  </action>
  <action name="actionToolBar">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/img/build_FILL0_wght400_GRAD0_opsz24.svg</normaloff>:/img/build_FILL0_wght400_GRAD0_opsz24.svg</iconset>
   </property>
   <property name="text">
    <string>工具栏</string>
   </property>
  </action>
  <action name="actionStatusBar">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/img/detector_status_FILL0_wght400_GRAD0_opsz24.png</normaloff>:/img/detector_status_FILL0_wght400_GRAD0_opsz24.png</iconset>
   </property>
   <property name="text">
    <string>状态栏</string>
   </property>
   <property name="toolTip">
    <string>状态栏</string>
   </property>
  </action>
  <action name="action_redo">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/img/redo_FILL0_wght400_GRAD0_opsz24.png</normaloff>:/img/redo_FILL0_wght400_GRAD0_opsz24.png</iconset>
   </property>
   <property name="text">
    <string>redo</string>
   </property>
   <property name="toolTip">
    <string>redo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+Y</string>
   </property>
  </action>
  <action name="actionSelectAll">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/img/select_all_FILL0_wght400_GRAD0_opsz24.png</normaloff>:/img/select_all_FILL0_wght400_GRAD0_opsz24.png</iconset>
   </property>
   <property name="text">
    <string>全选(&amp;A)</string>
   </property>
   <property name="toolTip">
    <string>全选</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="actionListNumber">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/img/format_list_numbered_FILL0_wght400_GRAD0_opsz24.png</normaloff>:/img/format_list_numbered_FILL0_wght400_GRAD0_opsz24.png</iconset>
   </property>
   <property name="text">
    <string>行号</string>
   </property>
   <property name="toolTip">
    <string>行号</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>关于(&amp;A)</string>
   </property>
   <property name="toolTip">
    <string>关于</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="actionFontcolor">
   <property name="text">
    <string>字体颜色</string>
   </property>
   <property name="toolTip">
    <string>字体颜色</string>
   </property>
  </action>
  <action name="actionbackcolor">
   <property name="text">
    <string>编辑器背景颜色</string>
   </property>
   <property name="toolTip">
    <string>编辑器背景颜色</string>
   </property>
  </action>
  <action name="actionfgbclolor">
   <property name="text">
    <string>字体背景颜色</string>
   </property>
   <property name="toolTip">
    <string>字体背景颜色</string>
   </property>
  </action>
  <action name="actionAddBookmark">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/img/shuqian.jpg</normaloff>:/img/shuqian.jpg</iconset>
   </property>
   <property name="text">
    <string>添加书签</string>
   </property>
  </action>
  <action name="actionShowBookmarks">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/img/书签.png</normaloff>:/img/书签.png</iconset>
   </property>
   <property name="text">
    <string>书签</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CodeEditor</class>
   <extends>QPlainTextEdit</extends>
   <header location="global">codeeditor.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="res.qrc"/>
 </resources>
 <connections/>
</ui>
